### PR TITLE
build: allow CMake to build and install shared libraries

### PR DESCRIPTION
**Pull request by mochaaP**
Signed-off-by: Zephyr Lykos <git@mochaa.ws>

### CODE REVIEW

**Comment by uael** [comment]
Thank you for your contribution!

Can you separate this into different commits ?
- One for the cmake minimum supported version update (only if really needed for the actual change).
- One for the actual change.
- (optional) one to reformat.

Although I'm not quite in favor of changing the format. It might not be perfect but it keep the changes history lean

**Comment by mochaaP** [comment]
Of course! Sorry for the format change, I have cmake-format configured in my editor. 😅

As of the CMake version requirement bump, it's needed for some new behaviors to work without explicitly stating them (which triggers a deprecation warning).

**Comment by mochaaP** [comment]
cc @uael ready to review ;)

**Comment by mochaaP** [comment]
Hey @uael,

I'm packaging this library into Fedora.  
If you could merge this and release a new version, we won't need to add a patch on our side.

Thanks!

File: CMakeLists.txt [hunk 1]
```diff
@@ -65,14 +66,9 @@ elseif (NOT MSVC)
...
-if (CMAKE_VERSION VERSION_LESS "3.1")
-  if (GCC)
-    set(CMAKE_C_FLAGS "-std=c99 ${CMAKE_C_FLAGS}")
-  endif ()
-else ()
-  set(CMAKE_C_STANDARD 99)
-endif ()
+set(CMAKE_C_STANDARD 99)
 
+option(BUILD_SHARED_LIBS "Build using shared libraries" ON)
```
Comment (uael) [inline CMakeLists.txt:71]: Is this a builtin option? I didn't catch any usage over the diff

**Review by uael — CHANGES_REQUESTED** [review]
Sorry for the late review, just one nit and we are good to go. Thank you for your contribution!

File: CMakeLists.txt (hunk 1 above)
Comment (mochaaP) [reply CMakeLists.txt:71]: Yes! See the change on L95, this will inherit to added targets.

Comment (mochaaP) [reply CMakeLists.txt:71]: https://github.com/uael/sv/pull/57/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aL95

Comment (mochaaP) [reply CMakeLists.txt:71]: https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html

**Comment by mochaaP** [comment]
More details replied in the review comments

**Review by uael — APPROVED** [review]
Sgtm!

**Pull request merged**